pkg/controller/nodeipam/ipam/cidrset/cidr_set_test.go

```diff
@@ -571,20 +571,19 @@ func TestGetBitforCIDR(t *testing.T) {
 		if err != nil {
 			t.Fatalf("unexpected error: %v for %v", err, tc.description)
 		}
-
 		got, err := cs.getIndexForCIDR(subnetCIDR)
 		if err == nil && tc.expectErr {
-			klog.Errorf("expected error but got null for %v", tc.description)
+			logger.Error(nil, "Expected error but got null", "description", tc.description)
```

Comment:
Looking at this again, I believe this test was broken. If something like this goes wrong in a test ("expected foo, got bar"), then the test should report it as a test failure with `t.Errorf`. Can you use that instead of contextual logging?

Reply:
While you are at it, use subtests (`t.Run(tc.description, func( ...)`) then the `Errorf` doesn't need to include the description.